src/components/ExerciseFilterModal.tsx

```diff
@@ -0,0 +1,195 @@
+                      </View>
+                    );
+                  })}
+                </View>
+                <View className="flex flex-grow flex-col ">
+                  {PRIMARYFILTER.slice(0, parseInt(PRIMARYFILTER.length / 2)).map((item, index) => {
+                    return (
+                      <View key={item} className="mt-4 flex flex-grow flex-row justify-start">
+                        <Text className="my-auto mr-2 font-bebas">{item}</Text>
+                        <Checkbox
+                          checked={selectedPrimary[index]}
+                          onPress={() => {
+                            setSelectedPrimary((prevArray) => {
+                              const tempArray = [...prevArray];
+                              tempArray[index] = !prevArray[index];
+                              return tempArray;
+                            });
+                          }}
+                        />
+                      </View>
+                    );
+                  })}
+                </View>
+              </View>
+              {/**Secondary: */}
+              <View className=" ml-4 mt-10 w-10/12 border-b border-slate-200  pb-2">
+                <Text className="font-bebas text-lg">Secondary:</Text>
+              </View>
+              <View className="mt-4 flex flex-row justify-between px-8">
+                <View className="flex flex-grow flex-col">
+                  {SECONDARYFILTER.slice(
+                    parseInt(SECONDARYFILTER.length / 2),
+                    SECONDARYFILTER.length
+                  ).map((item, index) => {
+                    return (
+                      <View key={item} className="mt-4 flex flex-grow flex-row justify-start">
+                        <Text className="my-auto mr-2 font-bebas">{item}</Text>
+                        <Checkbox
+                          checked={selectedSecondary[index + parseInt(SECONDARYFILTER.length / 2)]}
+                          onPress={() => {
+                            setSelectedSecondary((prevArray) => {
+                              const tempArray = [...prevArray];
+                              tempArray[index + parseInt(SECONDARYFILTER.length / 2)] =
+                                !prevArray[index + parseInt(SECONDARYFILTER.length / 2)];
+                              return tempArray;
+                            });
+                          }}
+                        />
+                      </View>
+                    );
+                  })}
+                </View>
+                <View className="flex flex-grow flex-col ">
+                  {SECONDARYFILTER.slice(0, parseInt(SECONDARYFILTER.length / 2)).map(
+                    (item, index) => {
+                      return (
+                        <View key={item} className="mt-4 flex flex-grow flex-row justify-start">
+                          <Text className="my-auto mr-2 font-bebas">{item}</Text>
+                          <Checkbox
```

Comment:
Can we align these checkboxes? I think if you just swap the order (render the checkbox before the text) it will align them and look better.

Reply:
actually, i swapped them intentionally because javascript tends round down the number so if im dealing with an odd number, the first half of the column gets fewer items than the second one.  Also aligning it can get tedious with the fflexbox.  A lot of if and then checks and also invisible Views might show to align them

Reply:
Swapping the elements and giving the text margin left of 2 does the trick, aligns the checkboxes vertically